src/scenarios/mauiandroid/pre.py

```diff
@@ -3,47 +3,58 @@
 '''
 import sys
 import os
+import requests
+import subprocess
 from zipfile import ZipFile
 from performance.logger import setup_loggers, getLogger
 from shutil import copyfile
+from shared import const
 from shared.precommands import PreCommands
-from shared.const import PUBDIR
 from argparse import ArgumentParser
+from test import EXENAME, MAUIVERSIONFILE
 
 setup_loggers(True)
 
-parser = ArgumentParser()
-parser.add_argument('--unzip', help='Unzip APK and report extracted tree', action='store_true', default=False)
+parser = ArgumentParser(add_help=False)
 parser.add_argument(
-        '--apk-name',
-        dest='apk',
-        required=True,
+        '-o', '--output',
+        dest='output_dir',
+        required=False,
         type=str,
-        help='Name of the APK to setup')
-args = parser.parse_args()
-
-if not os.path.exists(PUBDIR):
-    os.mkdir(PUBDIR)
-apkname = args.apk
-apknamezip = '%s.zip' % (apkname)
-if not os.path.exists(apkname):
-    getLogger().error('Cannot find %s' % (apkname))
-    exit(-1)
-if args.unzip:
-    if not os.path.exists(apknamezip):
-        copyfile(apkname, apknamezip)
-
-    with ZipFile(apknamezip) as zip:
-        zip.extractall(os.path.join('.', PUBDIR))
-
-    assets_dir = os.path.join(PUBDIR, 'assets')
-    assets_zip = os.path.join(assets_dir, 'assets.zip')
-    with ZipFile(assets_zip) as zip:
-        zip.extractall(assets_dir)
-
-    os.remove(assets_zip)
-else:
-    copyfile(apkname, os.path.join(PUBDIR, apkname))
-
-
-
+        help='capture of the output directory')
+args, unknown_args = parser.parse_known_args()
+
+# Download what we need
+with open ("MauiNuGet.config", "wb") as f:
+    f.write(requests.get(f'https://raw.githubusercontent.com/dotnet/maui/net7.0/NuGet.config', allow_redirects=True).content)
+
+precommands = PreCommands()
+precommands.install_workload('maui', ['--configfile', 'MauiNuGet.config'])
+
+# Setup the Maui folder
+precommands.new(template='maui',
+                output_dir=const.APPDIR,
+                bin_dir=const.BINDIR,
+                exename=EXENAME,
+                working_directory=sys.path[0],
+                no_restore=False)
+
+# Build the APK
+precommands.execute(['--no-restore', '--source', 'MauiNuGet.config'])
+
+# Remove the aab files as we don't need them, this saves space
+output_file_partial_path = f"com.companyname.{str.lower(EXENAME)}"
+if args.output_dir:
+    output_file_partial_path = os.path.join(args.output_dir, output_file_partial_path)
+
+os.remove(f"{output_file_partial_path}-Signed.aab")
+os.remove(f"{output_file_partial_path}.aab")
+
+# Copy the MauiVersion to a file so we have it on the machine
+result = subprocess.run(['powershell', '-Command', rf'Get-ChildItem .\{const.APPDIR}\obj\Release\net7.0-android\android-arm64\linked\Microsoft.Maui.dll | Select-Object -ExpandProperty VersionInfo | Select-Object ProductVersion | Select-Object -ExpandProperty ProductVersion'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
```

Comment:
I may be wrong as I don't use Powershell much, but I think you can remove the `Select-Object ProductVersion` since all it is doing is trimming the object down to one property, when you can just expand the property on the original object